src/DatabaseCache.php

```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
```

Comment:
I am surprised - if you want strict type checking you have to specify it in every file - see https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.strict 

I have the feeling that there is a tacit convention not to use this declaration?

Reply:
You don't "have to" specify it in every file, you specify it where you want strict types (no implicit typecasting / coercing of types).

It's true that for Kodus projects, we have generally not used these declarations, but there is no convention of ours that states we can't use it, if it makes sense.

On the PR for updating kodus/file-cache I suggest an alternative approach, that we could consider instead:
> I'm actually wondering if we are doing a bit of overkill by asserting the TypeError being thrown. If for some reason we would forget the typehints on the parameters, then as long as we implement the interface correctly and adhere to Liskov's Substitution Principle, I think our implementation could be considered correct. In other words, if our implementation can handle an integer key, that doesn't conflict with the interface, as long as it can handle the string keys required by the interface...

Let's revisit this on tomorrows stand up :-)

Reply:
https://github.com/kodus/file-cache/pull/11#issuecomment-1216475219